sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -189,15 +190,37 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
-        if self.freq == "auto" and fh.freq is None:
-            # when freq cannot be interpreted from ForecastingHorizon
-            raise ValueError(
-                f"Error in {self.__class__.__name__}, "
-                f"could not interpret freq, "
-                f"try passing freq in model initialization"
-            )
-
-        self._freq = fh.freq if self.freq == "auto" else self.freq
+        # A. freq is not auto {use this}
+        # B. freq is auto
+        #     B1. freq is infered {use this}
+        #     B2. freq is not infered
+        #         B2.1. y is date-like {use "D"}
+        #         B2.2. y is int-like  {use 1}
+        #         B2.3. other indexes; this part of code is never reached
+
+        if self.freq != "auto":
+            # A
+            self._freq = self.freq
+        else:
+            # B
+            if fh.freq:
+                # B1
+                self._freq = fh.freq
+            else:
+                # B2
+                if isinstance(y.index, (pandas.PeriodIndex, pandas.DatetimeIndex)):
+                    # B.2.1
+                    self._freq = "D"
+                elif isinstance(y.index, (pandas.Index, pandas.RangeIndex)):
```

Comment:
Since `y` is a `pandas.Series` object, is it not guaranteed that `isinstance(y.index, pandas.Index)` is `True` always?

Reply:
yes, you are right that will always be the case, so the condition here becomes irrelevant, only an `else` here would have done the job and it would also be safe because the code reaches here only if the index is date-like or int-like, an error is raised earlier for any other index.